### PR TITLE
Enhancement/wordcloud scrollbar

### PR DESCRIPTION
**Pull request by Quasar985**
### Prerequisites

- [x] Reviewed the [checklist](CHECKLIST.md)

- [x] Reviewed feedback from the "Sonar Cloud" bot. Note that you have to wait
    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change

Added scroll bar to the "Generate Content" section of the word cloud view. Also altered height of pane that contains the contents of word cloud to be based on the window size, instead of being a constant value.

### Alternate Designs

It was considered to keep the height as a constant value, however doing so would result in an inconsistent scale of the content page depending on the screen's resolution.

### Why Should This Be In Core?

Fixes issue where, depending on the user's screen resolution, the button to generate the word cloud is inaccessible.

### Benefits

Fixes issue related to word cloud generation being inaccessible.

### Possible Drawbacks

The functionality to dynamically set the height of the pane may need tweaking if the word cloud view undergoes major layout changes.

### Verification Process

Verified visually, by changing the window size, and by moving constellation to different monitors.

### Applicable Issues

https://github.com/constellation-app/constellation/issues/1961

### CODE REVIEW

File: CoreWordCloudView/src/au/gov/asd/tac/constellation/views/wordcloud/ui/WordCloudPane.java [hunk 1]
```diff
@@ -231,26 +236,30 @@ public WordCloudPane(final WordCloudController controller) {
         noWord.setMaxSize(0, 0);
     }
     
+    public void setEverythingHeight(int h) {
```
Comment (Delphinus8821) [inline CoreWordCloudView/src/au/gov/asd/tac/constellation/views/wordcloud/ui/WordCloudPane.java:239]: parameter final

**Review by Delphinus8821 — CHANGES_REQUESTED** [review]
A couple of things:
If you make the pane narrower than it expects, it cuts off the generate button even when there is room below it to expand to 
image

If you change your windows setting scaling to something quite zoomed in, the generate tab is still off of the screen. 
image

File: CoreWordCloudView/src/au/gov/asd/tac/constellation/views/wordcloud/ui/WordCloudTopComponent.java [hunk 2]
```diff
@@ -91,9 +90,14 @@ public WordCloudTopComponent() {
             final Scene scene = new Scene(wordCloudPane);
             scene.getStylesheets().addAll(JavafxStyleManager.getMainStyleSheet());
             panel.setScene(scene);
+
+            // Update word cloud pane's size when window size changes
+            scene.heightProperty().addListener((obv, oldVal, newVal) -> {
+                wordCloudPane.setContentHeight((int) (double) newVal);
```
Comment (antares1470) [inline CoreWordCloudView/src/au/gov/asd/tac/constellation/views/wordcloud/ui/WordCloudTopComponent.java:96]: I think only one of these type casts needs to occur

**Review by antares1470 — CHANGES_REQUESTED** [review]
Not sure if I'm a fan of the large gap being left at the top as it makes the Generate Word Cloud section look a little messy with its positioning. If possible, it would be good to have it as it was before with the panel showing the actual word cloud only displaying when there is something to be displayed.

wordCloudPanePositioning

**Comment by Delphinus8821** [comment]
> Not sure if I'm a fan of the large gap being left at the top as it makes the Generate Word Cloud section look a little messy with its positioning. If possible, it would be good to have it as it was before with the panel showing the actual word cloud only displaying when there is something to be displayed.
> 


I agree. It also looks a bit odd when there are not many words in the word cloud 

image

**Review by antares1470 — CHANGES_REQUESTED** [review]
On my laptop, once you generate a word cloud, the view doesn't fit horizontally within the default dimensions of the view space (including the Generate Word Cloud section which prior to generating a word didn't suffer from that issue). Perhaps an addition of a horizontal scroll bar would be useful (or better if possible, fitting the view to the default width of the view).

wordCloudHorizontal

**Review by Delphinus8821 — APPROVED** [review]

**Review by Delphinus8821 — APPROVED** [review]

**Review by antares1470 — APPROVED** [review]

**Pull request merged**